### PR TITLE
[Tests] Streamline

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | --
| License       | MIT

Extracted from https://github.com/symfony/symfony/pull/52157

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Tests/Controller/TemplateControllerTest.php [hunk 1]
```diff
@@ -32,13 +32,23 @@ public function testTwig()
...
+
+    public function testNoTwigInvokeMethod()
+    {
+        $controller = new TemplateController();
+
+        $this->expectException(\LogicException::class);
+        $this->expectExceptionMessage('You cannot use the TemplateController if the Twig Bundle is not available.');
+
         $controller('mytemplate')->getContent();
     }
```
Comment (OskarStark) [inline src/Symfony/Bundle/FrameworkBundle/Tests/Controller/TemplateControllerTest.php:53]: We should test both ways independently

File: src/Symfony/Bundle/SecurityBundle/Tests/Functional/SecurityRoutingIntegrationTest.php [hunk 2]
```diff
@@ -125,8 +125,7 @@ public function testInvalidIpsInAccessControl()
         $this->expectException(\LogicException::class);
         $this->expectExceptionMessage('The given value "256.357.458.559" in the "security.access_control" config option is not a valid IP address.');
 
-        $client = $this->createClient(['test_case' => 'StandardFormLogin', 'root_config' => 'invalid_ip_access_control.yml']);
-        $client->request('GET', '/unprotected_resource');
+        $this->createClient(['test_case' => 'StandardFormLogin', 'root_config' => 'invalid_ip_access_control.yml']);
```
Comment (OskarStark) [inline src/Symfony/Bundle/SecurityBundle/Tests/Functional/SecurityRoutingIntegrationTest.php:128]: Exception is thrown while creating the Client, no need to call `request()`

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php [hunk 3]
```diff
@@ -342,7 +342,7 @@ public function testPhpinfoActionWithProfilerDisabled()
         $twig = $this->createMock(Environment::class);
 
         $controller = new ProfilerController($urlGenerator, null, $twig, []);
-        $controller->phpinfoAction(Request::create('/_profiler/phpinfo'));
+        $controller->phpinfoAction();
```
Comment (OskarStark) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php:345]: `phpInfoAction` has no parameter

File: src/Symfony/Component/BrowserKit/Tests/AbstractBrowserTest.php
Comment (OskarStark) [inline src/Symfony/Component/BrowserKit/Tests/AbstractBrowserTest.php:1]: An exception is expected, the a `assertNull` is not needed

File: src/Symfony/Component/Cache/Tests/Adapter/RedisAdapterSentinelTest.php [hunk 4]
```diff
@@ -48,7 +48,7 @@ public function testExceptionMessageWhenFailingToRetrieveMasterInformation()
     {
         $hosts = getenv('REDIS_SENTINEL_HOSTS');
         $dsn = 'redis:?host['.str_replace(' ', ']&host[', $hosts).']';
-        $this->expectException(\Symfony\Component\Cache\Exception\InvalidArgumentException::class);
+        $this->expectException(InvalidArgumentException::class);
```
Comment (OskarStark) [inline src/Symfony/Component/Cache/Tests/Adapter/RedisAdapterSentinelTest.php:51]: already imported

File: src/Symfony/Component/Console/Tests/ApplicationTest.php [hunk 5]
```diff
@@ -1536,7 +1536,7 @@ public function testRunWithErrorAndDispatcher()
 
         $tester = new ApplicationTester($application);
         $tester->run(['command' => 'dym']);
-        $this->assertStringContainsString('before.dym.error.after.', $tester->getDisplay(), 'The PHP Error did not dispached events');
+        $this->assertStringContainsString('before.dym.error.after.', $tester->getDisplay(), 'The PHP Error did not dispatched events');
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/ApplicationTest.php]:
```suggestion
        $this->assertStringContainsString('before.dym.error.after.', $tester->getDisplay(), 'The PHP error did not dispatch events');
```

File: src/Symfony/Component/Console/Tests/ApplicationTest.php [hunk 6]
```diff
@@ -1553,7 +1553,7 @@ public function testRunDispatchesAllEventsWithError()
...
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/ApplicationTest.php]:
```suggestion
        $this->assertStringContainsString('before.dym.error.after.', $tester->getDisplay(), 'The PHP error did not dispatch events');
```

**Comment by OskarStark** [comment]
Ready to merge from my side

**Review by stof — APPROVED** [review]

**Review by yceruto — APPROVED** [review]

**Comment by OskarStark** [comment]
@nicolas-grekas as PRs get bigger and bigger, can you please merge/upmerge this one? Thanks

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 7]
```diff
@@ -1017,16 +1017,15 @@ public function testColumnStyle()
...
-        $table
-            ->setHeaders(['ISBN', 'Title', 'Author', 'Price'])
-            ->setRows([
-                ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
-            ]);
 
-        $table->render();
+        $table->setRows([
+            ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
+        ]);
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]:
```suggestion
        ]);
        $table->render();
```

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 8]
```diff
@@ -1017,15 +1017,15 @@ public function testColumnStyle()
 
     public function testThrowsWhenTheCellInAnArray()
     {
+        $table = new Table($output = $this->getOutputStream());
+        $table->setHeaders(['ISBN', 'Title', 'Author', 'Price']);
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]:
```suggestion
        $table = new Table($output = $this->getOutputStream());
        $table
            ->setHeaders(['ISBN', 'Title', 'Author', 'Price'])
            ->setRows([
                ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
            ]);
```

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 9]
```diff
@@ -1017,15 +1017,19 @@ public function testColumnStyle()
...
             ->setRows([
                 ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
             ]);
 
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('A cell must be a TableCell, a scalar or an object implementing "__toString()", "array" given.');
+
+        $table->setRows([
+            ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
+        ]);
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]:
```suggestion
```

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 10]
```diff
@@ -1017,15 +1017,16 @@ public function testColumnStyle()
 
     public function testThrowsWhenTheCellInAnArray()
     {
-        $this->expectException(InvalidArgumentException::class);
-        $this->expectExceptionMessage('A cell must be a TableCell, a scalar or an object implementing "__toString()", "array" given.');
         $table = new Table($output = $this->getOutputStream());
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php]:
```suggestion
        $table = new Table($this->getOutputStream());
```

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 11]
```diff
@@ -1017,15 +1017,16 @@ public function testColumnStyle()
 
     public function testThrowsWhenTheCellInAnArray()
     {
-        $this->expectException(InvalidArgumentException::class);
-        $this->expectExceptionMessage('A cell must be a TableCell, a scalar or an object implementing "__toString()", "array" given.');
-        $table = new Table($output = $this->getOutputStream());
+        $table = new Table($this->getOutputStream());
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php:1020]: Unused variable `$output`

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php [hunk 12]
```diff
@@ -1017,15 +1017,16 @@ public function testColumnStyle()
...
         $table
             ->setHeaders(['ISBN', 'Title', 'Author', 'Price'])
             ->setRows([
                 ['99921-58-10-7', [], 'Dante Alighieri', '9.95'],
             ]);
 
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('A cell must be a TableCell, a scalar or an object implementing "__toString()", "array" given.');
+
         $table->render();
```
Comment (OskarStark) [inline src/Symfony/Component/Console/Tests/Helper/TableTest.php:1030]: @stof it seems, that this is no dead code, as the exception is raised when calling `render()` and not when calling `setRows()` 👍

**Comment by OskarStark** [comment]
Fabbot looks unrelated

**Comment by nicolas-grekas** [comment]
Thank you @OskarStark.

File: src/Symfony/Component/Console/Tests/Helper/TableTest.php (hunk 12 above)
Comment (stof) [reply src/Symfony/Component/Console/Tests/Helper/TableTest.php:1030]: then in the other PR, keep `setRows` in initialization

**Pull request merged**